src/chip.cs

```diff
@@ -0,0 +1,1767 @@
+		string OpenPairingWindow (nuint duration, nuint discriminator, nuint setupPin, [NullAllowed] out NSError error);
+
+		[Export ("isActive")]
+		bool IsActive { get; }
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject), Name="CHIPCluster")]
+	[DisableDefaultCtor]
+	interface ChipCluster
+	{
+		[Export ("initWithDevice:endpoint:queue:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	delegate void ChipResponseHandler ([NullAllowed] NSError error, [NullAllowed] NSDictionary data);
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPApplicationBasic")]
+	[DisableDefaultCtor]
+	interface ChipApplicationBasic
+	{
+		[Export ("initWithDevice:endpoint:queue:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeVendorNameWithResponseHandler:")]
+		void ReadAttributeVendorName (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeVendorIdWithResponseHandler:")]
+		void ReadAttributeVendorId (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeApplicationNameWithResponseHandler:")]
+		void ReadAttributeApplicationName (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeProductIdWithResponseHandler:")]
+		void ReadAttributeProductId (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeApplicationIdWithResponseHandler:")]
+		void ReadAttributeApplication (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeCatalogVendorIdWithResponseHandler:")]
+		void ReadAttributeCatalogVendorId (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeApplicationSatusWithResponseHandler:")]
+		void ReadAttributeApplicationSatus (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
+	}
+
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(ChipCluster), Name="CHIPBarrierControl")]
+	[DisableDefaultCtor]
+	interface ChipBarrierControl
+	{
+
+		[Export ("initWithDevice:endpoint:queue:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
```

Comment:
When should we use this ResultTypeName with Async?

Reply:
It is related when the handler returns more than one value:
```csharp
delegate void ChipResponseHandler ([NullAllowed] NSError error, [NullAllowed] NSDictionary data);
```

The generator will tell you.